src/Neo/Network/P2P/Payloads/NotValidBefore.cs

```diff
@@ -0,0 +1,46 @@
+using Neo.IO;
+using Neo.Json;
+using Neo.Persistence;
+using Neo.SmartContract.Native;
+using System.IO;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class NotValidBefore : TransactionAttribute
+    {
+        /// <summary>
+        /// Indicates that the transaction is not valid before this height.
+        /// </summary>
+        public uint Height;
+
+        public override TransactionAttributeType Type => TransactionAttributeType.NotValidBefore;
+
+        public override bool AllowMultiple => false;
+
+        public override int Size => base.Size +
+            sizeof(uint); // Height.
+
+        protected override void DeserializeWithoutType(ref MemoryReader reader)
+        {
+            Height = reader.ReadUInt32();
+        }
+
+        protected override void SerializeWithoutType(BinaryWriter writer)
+        {
+            writer.Write(Height);
+        }
+
+        public override JObject ToJson()
+        {
+            JObject json = base.ToJson();
+            json["height"] = Height;
+            return json;
+        }
+
+        public override bool Verify(DataCache snapshot, Transaction tx)
+        {
+            var block_height = NativeContract.Ledger.CurrentIndex(snapshot);
```

Comment:
Add any valid range?

Reply:
Agree. This PR depends on https://github.com/neo-project/neo/pull/2812, that's why the first commit (fdc3934009022a2bb88611a48a8d802aec64142e) is here. I'll rebase it after https://github.com/neo-project/neo/pull/2812 merge.

Reply:
I would like to prevent to keep a transaction for the next 10 years

Reply:
How long you keep it is VUB, shouldn't be a problem.